pkg/v1/tkg/managementcomponents/management_component_install.go

```diff
@@ -48,18 +61,207 @@ type ManagementComponentsInstallOptions struct {
+	pkgiName := fmt.Sprintf("tanzu-%s", addonsManagerName)
+	addonSecretName := generateAddonSecretName(clusterName, addonsManagerName)
+
+	err := pauseAddonSecretReconciliation(context.TODO(), clusterClient, addonSecretName, namespace)
+	if err != nil {
+		return err
+	}
+
+	err = pausePackageInstallReconciliation(context.TODO(), clusterClient, pkgiName, namespace)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// NoopDeletePackageInstall sets spec.noopdelete = true before deleting the package install
+func NoopDeletePackageInstall(clusterClient clusterclient.Client, addonName, namespace string) error {
+	log.Infof("Deleting %s/%s with noopdelete", namespace, addonName)
+	pkgiName := fmt.Sprintf("tanzu-%s", addonName)
+
+	jsonPatch := []map[string]interface{}{
+		{
+			"op":    "add",
+			"path":  "/spec/noopDelete",
+			"value": true,
+		},
+	}
+	payloadBytes, err := json.Marshal(jsonPatch)
+	if err != nil {
+		return errors.Wrap(err, "unable to generate json patch")
+	}
+	pkgi := &kappipkg.PackageInstall{}
+	err = clusterClient.PatchResource(pkgi, pkgiName, namespace, string(payloadBytes), types.JSONPatchType, nil)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to patch %s packageinstall", pkgiName)
+	}
+
+	pkgi.Name = pkgiName
+	pkgi.Namespace = namespace
+	err = clusterClient.DeleteResource(pkgi)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrapf(err, "failed to delete PackageInstall resource %s", pkgiName)
+	}
+	return nil
+}
+
+func DeleteAddonSecret(clusterClient clusterclient.Client, clusterName, addonName, namespace string) error {
+	addonSecret := &corev1.Secret{}
+	addonSecret.Name = generateAddonSecretName(clusterName, addonName)
+	addonSecret.Namespace = constants.TkgNamespace
+	log.Infof("Deleting %s/%s secret", addonSecret.Namespace, addonSecret.Name)
+	err := clusterClient.GetResource(addonSecret, addonSecret.Name, addonSecret.Namespace, nil, nil)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if controllerutil.ContainsFinalizer(addonSecret, addonFinalizer) {
+		controllerutil.RemoveFinalizer(addonSecret, addonFinalizer)
+	}
+
+	err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		return clusterClient.UpdateResource(addonSecret, addonSecret.Name, addonSecret.Namespace)
+	})
+
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	err = clusterClient.DeleteResource(addonSecret)
+	if apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete addon addonSecret %s", addonSecret)
+	}
+	return nil
+}
+
+func addonsPackageIsMovingRepositories(clusterClient clusterclient.Client, clusterName, addonName, namespace string) (bool, error) {
+	// We can check if the addons package has been installed from the core repository (https://gitlab.eng.vmware.com/core-build/tkg-packages/-/tree/main/core/addons-manager)
```

Comment:
remove references to internal repositories

Reply:
ack

Reply:
fixed